x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/maintenance_window_callout/api.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MaintenanceWindow } from '@kbn/alerting-plugin/common/maintenance_window';
+import { KibanaServices } from '../../../../common/lib/kibana';
+import { GET_ACTIVE_MAINTENANCE_WINDOWS_URL } from './constants';
+
+export const fetchActiveMaintenanceWindows = async (
```

Comment:
While the data looks isolated to this component so far it's easy to imagine a situation when the same data is required at some other place. We have already `x-pack/plugins/security_solution/public/detection_engine/rule_management/api/api.ts` and `x-pack/plugins/security_solution/public/detection_engine/rule_management/logic` containing React Query hooks. To keep data related logic at one place it's wise to move `fetchActiveMaintenanceWindows` and `useFetchActiveMaintenanceWindows` to the mentioned above folders.

Reply:
I also thought about moving these to a level above, but wasn't sure if it would be worth it. Then I had a conversation with @banderror about a similar question in another branch. He has referred me to our [Kibana Dev Principles doc](https://github.com/elastic/engineering/blob/main/kibana_dev_principles.md#dont-share-code-prematurely) where it's suggested that we should avoid sharing code prematurely. So I concluded that it'd be fine to co-locate it within the same component directory.  

Also the Response Ops team is going to take over this code from us and move it to their plugin before 8.9. So it probably doesn't really matter where we put it for now 😃 

wdyt?